### PR TITLE
Added Google Cloud Storage plugin

### PR DESCRIPTION
**Pull request by nkottary**
* All methods unit tested for text and binary files.
* Tested from Juliabox by uploading files and logging out and logging in.
* Setup instructions in `info.md` file.

### CODE REVIEW

File: engine/src/juliabox/plugins/bucket_gs/impl_gs.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    @staticmethod
+    def pull(bucket, local_file, metadata_only=False):
+        objname = os.path.basename(local_file)
+        if metadata_only:
+            k = None
+            try:
+                k = JBoxGS.connect().objects().get(bucket=bucket,
+                                                   object=objname).execute()
+            except:
+                pass
```
Comment (tanmaykm) [inline engine/src/juliabox/plugins/bucket_gs/impl_gs.py]: may be better not to silently ignore exception?

File: engine/src/juliabox/plugins/bucket_gs/impl_gs.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                                                       object=objname)
+
+            fh = open(local_file, "wb")
+            downloader = MediaIoBaseDownload(fh, req, chunksize=1024*1024)
+            done = False
+            while not done:
+                try:
+                    _, done = downloader.next_chunk()
+                except:
+                    break
```
Comment (tanmaykm) [inline engine/src/juliabox/plugins/bucket_gs/impl_gs.py]: should probably let the exception be raised, and close `fh` in the `finally` clause

**Comment by tanmaykm** [comment]
Thanks. This looks good!

**Comment by nkottary** [comment]
I have edited the code to check the exceptions.

**Comment by nkottary** [comment]
A `pull_from_bucketstore` is done for a user for the first time even when there is nothing in the bucket.  I saw that a `None` was expected in this scenario hence the `return None` in the `except` block.

File: engine/src/juliabox/plugins/bucket_gs/impl_gs.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        else:
+            req = JBoxGS.connect().objects().get_media(bucket=bucket,
+                                                       object=objname)
+
+            fh = open(local_file, "wb")
+            downloader = MediaIoBaseDownload(fh, req, chunksize=1024*1024)
+            done = False
+            while not done:
+                _, done = downloader.next_chunk()
+            fh.close()
```
Comment (tanmaykm) [inline engine/src/juliabox/plugins/bucket_gs/impl_gs.py]: close file handle if download fails and delete the file. got missed out in the latest change.

````
fh = open(...)
done = False
try:
   ...
finally:
   fh.close()
   if not done:
       os.remove(local_file)  
````

**Comment by nkottary** [comment]
I have made the changes in a new PR #359

**Pull request closed without merging**